src/app/services/data.service.ts

```diff
@@ -215,16 +261,18 @@ export class DataService {
         // Calculate availability of this component
         this.calculateAvailability(componentId);
       });
-      // We are now fully loaded and can display the data
-      this._loadingFinished.next(true);
+      // Check if we are now fully loaded and can display the data
+      componentsDone = true;
+      if (componentsDone && updatesDone) {
+        this._loadingFinished.next(true);
+      }
     });
   }
 
   private calculateAvailability(component: ComponentId): void {
     const statusList = this.componentStatusByDay.get(component);
     if (!statusList) {
-      // TODO Error properly
-      console.log("Found a component with missing daily status list?");
+      console.error("Found a component with missing daily status list?");
```

Comment:
No `toastr`? Is this an "internal" error?

Reply:
I mentioned this in a comment above. The `DataService` would have to rely on the `ToastrService` being loaded before it could make use of ntx-toastr. I'd have to add this dependency to the service configuration and test it. For the moment, I'm busy trying to find out why, after the recent changes, loading never finishes and the site is stuck displaying the loading animation.